### DISCUSSION TITLE
Integration with Unity

### DISCUSSION THREAD

**Discussion by Popugames** [WalletConnect/walletconnect-monorepo#2593] [Developer Support]
I imported your C# SDK into Unity but it uses certain .NET libraries that are not compatible with Unity. After compiling with IL2CPP, nothing works anymore.

It only works in Editor, after compilation it doesn't work anymore.

We need a special SDK for Unity that does not use .NET libraries.

**Comment by finessevanes** [comment]
Hey @Popugames I've reached out to our Unity team to take a look!

**Comment by gigajuwels** [comment]
Hi @Popugames can you please provide the error message you are getting when using IL2CPP and which libraries you are having trouble with?

**Reply by Popugames** [reply]
Hi @gigajuwels, Thank you for reply, 

`TimeoutException: Timeout of 20000 exceeded
WalletConnectSharp.Common.Utils.Extensions.WithTimeout[T] (System.Threading.Tasks.Task`1[TResult] task, System.Int32 timeout, System.String message) (at Assets/Wallet Connect v2/Core Modules/WalletConnectSharp.Common/Utils/Extensions.cs:98)
WalletConnectSharp.Core.Controllers.Subscriber.RpcSubscribe[TR] (System.String topic, WalletConnectSharp.Core.Models.Relay.ProtocolOptions relay) (at Assets/Wallet Connect v2/WalletConnectSharp.Core/Controllers/Subscriber.cs:319)
WalletConnectSharp.Core.Controllers.Subscriber.Subscribe (System.String topic, WalletConnectSharp.Core.Models.Relay.SubscribeOptions opts) (at Assets/Wallet Connect v2/WalletConnectSharp.Core/Controllers/Subscriber.cs:543)
WalletConnectSharp.Core.Controllers.Pairing.Create () (at Assets/Wallet Connect v2/WalletConnectSharp.Core/Controllers/Pairing.cs:240)
WalletConnectSharp.Sign.Engine.Connect (WalletConnectSharp.Sign.Models.Engine.ConnectOptions options) (at Assets/Wallet Connect v2/WalletConnectSharp.Sign/Engine.cs:209)
Connect.Execute () (at Assets/Connect.cs:62)
Connect.Start () (at Assets/Connect.cs:26)`

**Comment by mustafadervisglu** [comment]
did you solve this error ? @gigajuwels
